### PR TITLE
add the expand query parameter

### PR DESCRIPTION
**Pull request by frennkie**
As discussed in #197 it make sense to have the query parameter `$expand` (mostly adapted from the $select parameter).

The `convert_case`  breaks the value of e.g. `microsoft.graph.eventMessage/event` (it results in `microsofrapventMessagvent`. I only tested this on `MSGraphProtocol`, so this will most likely need to be checked fixed for `MSOffice365Protocol`

### CODE REVIEW

**Comment by frennkie** [comment]
@janscas Feel free to edit or even discard if you have already done this separately/differently.. :-D

File: O365/utils/utils.py [hunk 1]
```diff
@@ -321,15 +321,37 @@ def __init__(self, attribute=None, *, protocol):
...
+    @fluent
+    def expand(self, *relationships):
+        """ Adds the relationships (e.g. "microsoft.graph.eventMessage/event" or
+        "attachments") that should be expanded to the $expand parameter
+
+        :param str relationships: the relationships tuple to expand.
+        :rtype: Query
+        """
+
+        for relationship in relationships:
```
Comment (janscas) [inline O365/utils/utils.py]: To control whether to add 'microsoft.graph.eventMessage' if the protocol is MSGrapProtocol or the 'Microsoft.OutlookServices.EventMessage' if you are using Outlook rest api, you have to call self.protocol.get_service_keyword. I'll implement this

**Comment by janscas** [comment]
@frennkie This PR is a Draft so I can not merge it nor use the code. 
I'll commit the changes that include some of your code and some modifications

**Comment by frennkie** [comment]
Feel free.

I was not aware that you can not edit this Draft PR.. First time I'm using it. :-D

**Comment by frennkie** [comment]
@janscas I did a work-in-progress implementation of auto parse too.. Might be a reference for you: https://github.com/frennkie/python-o365/tree/auto-parse-event?files=1

**Comment by janscas** [comment]
> did a work-in-progress implementation of auto parse too

What do you mean by 'auto parse'?

**Comment by frennkie** [comment]
Have a look at it... If $expand was given then I parse the Event and add it the event attribute of Message.

**Comment by janscas** [comment]
> Have a look at it... If $expand was given then I parse the Event and add it the event attribute of Message.

ok, this is already done in the commit

**Pull request closed without merging**